### PR TITLE
`no-descending-specificity` - added exception for custom property sets

### PR DESCRIPTION
**Pull request by woble**
> Which issue, if any, is this issue related to?

#1999 

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

File: src/rules/no-descending-specificity/index.js [hunk 1]
```diff
@@ -35,6 +35,9 @@ export default function (actual) {
         // Ignore `.selector, { }`
         if (trimSelector === "") { return }
 
+        // Ignore custom property set `--foo: {};`
+        if (/^--.+:$/.test(trimSelector)) { return }
```
Comment (evilebottnawi) [inline src/rules/no-descending-specificity/index.js]: we have util `isCustomPropertySet` https://github.com/stylelint/stylelint/blob/master/src/utils/isCustomPropertySet.js for this

**Comment by evilebottnawi** [comment]
@woble squash your comment in one for more convenient review

File: src/rules/no-descending-specificity/README.md [hunk 2]
```diff
@@ -105,3 +105,9 @@ a {}
   #baz a {}
 }
 ```
+
+```css
```
Comment (jeddy3) [inline src/rules/no-descending-specificity/README.md]: There's no need to add this to the README as it's an edge-case. We try to keep the READMEs as simple as possible and only show common patterns. Can you please remove this pattern.

File: src/rules/no-descending-specificity/index.js [hunk 3]
```diff
@@ -35,6 +36,9 @@ export default function (actual) {
         // Ignore `.selector, { }`
         if (trimSelector === "") { return }
 
+        // Ignore custom property set `--foo: {};`
+        if (isCustomPropertySet(rule)) { return }
```
Comment (jeddy3) [inline src/rules/no-descending-specificity/index.js]: Can you please move this line to between line 31 and 32. We'd want to return _as early as possible_, i.e. before doing anything to the selector.

**Comment by jeddy3** [comment]
LGTM. @woble Thanks for making the changes.

**Comment by davidtheclark** [comment]
Changelog note added:

> -   Fixed: `no-descending-specificity` now ignores custom property sets.

**Pull request merged**